src/equations/acoustic_perturbation_2d.jl

```diff
@@ -104,51 +91,198 @@ function source_terms_convergence_test(u, x, t, equations::AcousticPerturbationE
+  # y-direction by -1.
+  # For every other boundary, we set the perturbed variables to zero and take the mean variables
+  # from the inner state
+  if direction == 3 # boundary in -y direction
+    if -0.05 <= x[1] <= 0.05 # Monopole
+      v1_prime = 0.0
+      v2_prime = p_prime = sin(2 * pi * t)
+
+      u_boundary = SVector(v1_prime, v2_prime, p_prime, u_inner[4], u_inner[5], u_inner[6],
+                           u_inner[7])
+    else # Wall
+      u_boundary = SVector(u_inner[1], -u_inner[2], u_inner[3], u_inner[4], u_inner[5], u_inner[6],
+                           u_inner[7])
+    end
+  else
+    u_boundary = SVector(0.0, 0.0, 0.0, u_inner[4], u_inner[5], u_inner[6], u_inner[7])
+  end
+
+  # Calculate boundary flux
+  if direction in (2, 4) # u_inner is "left" of boundary, u_boundary is "right" of boundary
+    flux = surface_flux_function(u_inner, u_boundary, orientation, equations)
+  else # u_boundary is "left" of boundary, u_inner is "right" of boundary
+    flux = surface_flux_function(u_boundary, u_inner, orientation, equations)
+  end
+
+  return flux
 end
 
 
 # Calculate 1D flux for a single point
 @inline function calcflux(u, orientation, equations::AcousticPerturbationEquations2D)
-  v1_prime, v2_prime, p_prime = u
-  @unpack v_mean, rho_mean, c_mean = equations
+  v1_prime, v2_prime, p_prime, v1_mean, v2_mean, c_mean, rho_mean = u
 
+  # Calculate flux for conservative state variables
   if orientation == 1
-    f1 = v_mean[1] * v1_prime + v_mean[2] * v2_prime + p_prime / rho_mean
+    f1 = v1_mean * v1_prime + v2_mean * v2_prime + p_prime / rho_mean
     f2 = zero(eltype(u))
-    f3 = c_mean^2 * rho_mean * v1_prime + v_mean[orientation] * p_prime
+    f3 = c_mean^2 * rho_mean * v1_prime + v1_mean * p_prime
   else
     f1 = zero(eltype(u))
-    f2 = v_mean[1] * v1_prime + v_mean[2] * v2_prime + p_prime / rho_mean
-    f3 = c_mean^2 * rho_mean * v2_prime + v_mean[orientation] * p_prime
+    f2 = v1_mean * v1_prime + v2_mean * v2_prime + p_prime / rho_mean
+    f3 = c_mean^2 * rho_mean * v2_prime + v2_mean * p_prime
   end
 
-  return SVector(f1, f2, f3)
+  # The rest of the state variables are actually variable coefficients, hence the flux should be
+  # zero. See https://github.com/trixi-framework/Trixi.jl/issues/358#issuecomment-784828762
+  # for details.
+  f4 = f5 = f6 = f7 = zero(eltype(u))
+
+  return SVector(f1, f2, f3, f4, f5, f6, f7)
 end
 
 
 function flux_lax_friedrichs(u_ll, u_rr, orientation, equations::AcousticPerturbationEquations2D)
   f_ll = calcflux(u_ll, orientation, equations)
   f_rr = calcflux(u_rr, orientation, equations)
 
-  v_ll = u_ll[orientation] + equations.v_mean[orientation]
-  v_rr = u_rr[orientation] + equations.v_mean[orientation]
-  c0 = equations.c_mean
-  speed = max(abs(v_ll), abs(v_rr)) + c0
+  # Calculate v = v_prime + v_mean
+  v_ll = u_ll[orientation] + u_ll[orientation + 3]
+  v_rr = u_rr[orientation] + u_rr[orientation + 3]
```

Comment:
```suggestion
  v_prime_ll = u_ll[orientation]
  v_prime_rr = u_rr[orientation]
  v_mean_ll = u_ll[orientation + 3]
  v_mean_rr = u_rr[orientation + 3]
  
  v_ll = v_prime_ll + v_mean_ll
  v_rr = v_prime_rr + v_mean_rr
```
I feel like that would make things a little clearer and easier to read (though more verbose).